client/app/components/home/home.controller.js

```diff
@@ -1,22 +1,32 @@
 class HomeController {
-  constructor($state, $document, $mdDialog, GroupService) {
+  constructor($state, $document, $mdDialog, GroupService, Authentication) {
     "ngInject";
     Object.assign(this, {
       $state,
       $document,
       $mdDialog,
-      GroupService
+      GroupService,
+      Authentication
     });
   }
 
   $onInit() {
-    this.GroupService.listMy().then((data) => {
-      if (data.length > 0) {
-        this.$state.go("group", { groupId: data[0].id });
-      } else {
-        this.openJoinGroupDialog();
-      }
-    });
+    if (angular.isUndefined(this.Authentication.data)) {
+      this.Authentication.update();
```

Comment:
Same problem here - remember that Authentication.update() returns a promise. The code that follows would need to be run in a `then` function.

But we can safely assume that `Authentication.data` is defined - the `hookProvider` checks that (and redirects to the login screen if the user is not logged in). There is no race condition there: the state only gets loaded when the [hookProvider did its job](https://github.com/yunity/foodsaving-frontend/blob/b840c3dfc6f06181603f676d4459ca9881aad71e/client/app/services/authentication/hook.js#L20)